feat: show manifests in image list

cdrage opened the pull request:
feat: show manifests in image list

### What does this PR do?

* Shows manifest in the image list
* Does NOT show any actions / be able to click (yet)
* Allows you to click and view details.

### Screenshot / video of UI

<!-- If this PR is changing UI, please include
screenshots or screencasts showing the difference -->

https://github.com/containers/podman-desktop/assets/6422176/e930329d-2a84-4c2b-b719-373f12f0b22c


### What issues does this PR fix or reference?

<!-- Include any related issues from Podman Desktop
repository (or from another issue tracker). -->

Closes https://github.com/containers/podman-desktop/issues/6529
Closes https://github.com/containers/podman-desktop/issues/7255

### How to test this PR?

<!-- Please explain steps to verify the functionality,
do not forget to provide unit/component tests -->

- [X] Tests are covering the bug fix or the new feature

1. Create a manifest through the GUI or `podman build --platform
   linux/arm64,linux/amd64 --manifest testmanifest123`
2. View the manifest in image list.

Signed-off-by: Charlie Drage <charlie@charliedrage.com>

cdrage commented:
FYI it is intentional that there are no actions shown / unable to click for more details / no delete button. As those are in separate PR's in the epic: https://github.com/containers/podman-desktop/issues/6526 specifically issues: #7041 and #7181

cdrage commented on packages/renderer/src/lib/image/image-utils.ts:
```diff
@@ -172,6 +172,11 @@ export class ImageUtils {
 
+    // If the image isManifest = true, we will add a badge to the image that simply says "manifest"
+    if (imageInfo.isManifest) {
+      badges.push({ label: 'manifest', color: { light: 'bg-purple-600', dark: 'bg-purple-600' } });
```
FYI unsure which colour we should use to indicate manifest, so just chose bg-purple-600 for now @deboer-tim @benoitf

deboer-tim reviewed (changes requested):
I think it's curious/annoying that manifests are images in the API, and makes it so that something like ImageColumnActions needs to be shared. :-/

I think this is a good place to go with the first PR, but:
- In the containers view we have "my-pod (pod)", but here you used a badge. We obviously didn't have badges when we did the containers view, but I think we need to decide with design the right way to identify 'different things in a table' is and make them consistent. For simplicity/consistency maybe I'd rather use '(manifest)' until we know.
- The icon should be different from a regular image.
- Manifests should have details pages (but can be a separate issue).
- We should allow them to expand/collapse with the images that are in them, but that is post #7209.

I don't want to hold this up too much, but I think we should at least have a direction on 1/2 and issue for 3 before we merge.

cdrage commented:
> I think it's curious/annoying that manifests are images in the API, and makes it so that something like ImageColumnActions needs to be shared. :-/
> 
> I think this is a good place to go with the first PR, but:
> 
> * In the containers view we have "my-pod (pod)", but here you used a badge. We obviously didn't have badges when we did the containers view, but I think we need to decide with design the right way to identify 'different things in a table' is and make them consistent. For simplicity/consistency maybe I'd rather use '(manifest)' until we know.
> * The icon should be different from a regular image.
> * Manifests should have details pages (but can be a separate issue).
> * We should allow them to expand/collapse with the images that are in them, but that is post [Table component support for parent/children #7209](https://github.com/containers/podman-desktop/issues/7209).
> 
> I don't want to hold this up too much, but I think we should at least have a direction on 1/2 and issue for 3 before we merge.

I totally agree, let's talk about it in the UX call. There are a lot of other missing pieces that should be worked on before we merge the "showing in list" support. 

For this PR I will:

- [ ] Add (manifest) similar to (my-pod) instead of a badge
- [ ] Different icon (will ask you or @ekidneyrh on advice what we should do)

I thought I had opened a "details" issue for manifest, but I had not. I've opened up issue https://github.com/containers/podman-desktop/issues/7255 and added it to the epic https://github.com/containers/podman-desktop/issues/6526

Maybe we should keep this as draft / WIP until we get the details page in too so we at least have the basic functionality of manifests.

cdrage commented:
This PR is ready for review with the exception of the new icon being added, which can be in a follow up PR.

I've made the following changes:
- New folder / files, based on how we do ImageDetails.svelte
- Added a BASIC manifest summary page, a more advanced summary page based on using inspectManifest will be added via issue: https://github.com/containers/podman-desktop/issues/7270
- Added tests

lstocchi commented on packages/renderer/src/lib/image/ImageDetailsHistory.svelte:
```diff
@@ -10,6 +10,7 @@ let history: string;
 
 onMount(async () => {
   // grab inspect result from the container
+  console.log(image.engineId, image.id);
```
to be removed

lstocchi reviewed (approved):
Tested and works as described. LGTM

cdrage commented:
Manifest icon has now been added and this PR is ready for another review @lstocchi @deboer-tim

deboer-tim reviewed (approved):
LGTM, worked well in testing.
